pkg/util/git.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/pkg/errors"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/config"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+const (
+	BranchRE = `master|release-([0-9]{1,})\.([0-9]{1,})(\.([0-9]{1,}))*$`
+	// TODO: Use "kubernetes" as DefaultGithubOrg when this is ready to turn on
+	DefaultGithubOrg      = "justaugustus"
+	DefaultGithubAuthRoot = "git@github.com:"
+	DefaultRemote         = "origin"
+)
+
+var (
+	KubernetesGitHubURL     = fmt.Sprintf("https://github.com/%s/kubernetes", DefaultGithubOrg)
+	KubernetesGitHubAuthURL = fmt.Sprintf("%s%s/kubernetes.git", DefaultGithubAuthRoot, DefaultGithubOrg)
+)
+
+func BranchExists(branch string) (bool, error) {
+	log.Printf("Verifying %s branch exists on the remote...", branch)
+
+	// Create the remote with repository URL
+	rem := git.NewRemote(
+		memory.NewStorage(),
+		&config.RemoteConfig{
+			Name: DefaultRemote,
+			URLs: []string{KubernetesGitHubURL},
+		},
+	)
+
+	log.Print("Fetching tags...")
+
+	// We can then use every Remote functions to retrieve wanted information
+	refs, err := rem.List(&git.ListOptions{})
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Would be useful to return an informative error here.

Reply:
Later: We could use something like wrapping errors with additional context to provide some sort of error stack.